sqlglot/dialects/tsql.py

```diff
@@ -708,6 +716,53 @@ def parse_range():
 
             return partition
 
+        def _parse_declare(self) -> exp.Declare | exp.Command:
+            index = self._index
+            try:
+                expressions = []
+                while True:
+                    if self._match(TokenType.PARAMETER, advance=False):
+                        var = self._parse_id_var()
+                        size = None
+                        value = None
+                        # moving past the AS keyword if it's there
+                        self._match_text_seq("AS")
+
+                        if self._match(TokenType.TABLE):
+                            table = self.expression(exp.Table, this=var)
+                            data_type = self._parse_schema(this=table)
+                        else:
+                            data_type = self._parse_types()  # Parse data type
+                            # Look for an optional size within parentheses
+                            if self._match(TokenType.L_PAREN):
+                                size = self._parse_bitwise()
+                                self._match(TokenType.R_PAREN)
```

Comment:
This is redundant, the parser already handles parameterized types. I think `_parse_type()` suffices?

Reply:
Let's also remove these comments - "Parse data type" doesn't really provide any useful insight into what's going on here.